core/dbt/deps/tarball.py

```diff
@@ -0,0 +1,249 @@
+        # self.tarfile_bin = self.get_tarfile()
+        # self.get_tarfile()
+        # self.tar_dir_name = self.resolve_tar_dir()
+
+    def get_version(self):
+        return None
+
+    def nice_version_name(self):
+        return '<tarball @ {}>'.format(self.tarball)
+
+    def get_tar_size(self):
+        return sum(x.size for x in self.tarfile_bin.getmembers())
+
+    def file_sha1(self, fp_in: str):
+        if type(fp_in) is str:
+            with open(fp_in, 'rb') as fp:
+                return hashlib.sha1(fp.read()).hexdigest()
+        elif hasattr(fp_in, 'read'):
+            return hashlib.sha1(fp_in.read()).hexdigest()
+        else:
+            raise ValueError()
+
+    def get_tarfile(self):
+        def validate_checksum(self, filepath):
+            if self.sha1:
+                checksum = self.file_sha1(filepath)
+                if not checksum == self.sha1:
+                    package_sha1_fail(checksum, self.sha1)
+
+        def validate_tarfile(self, filepath):
+            if not tarfile.is_tarfile(filepath):
+                msg = f"{filepath} is not a valid tarfile."
+                raise DependencyException(msg)
+
+        def handle_local_tar(self):
+            # pass through local tarfile path primarily for testing
+            validate_checksum(self, self.tarball)
+            validate_tarfile(self, self.tarball)
+            tar = tarfile.open(self.tarball, "r:gz")
+            self.tarfile_bin = tar
+
+        def handle_remote_tar(self):
+            with NamedTemporaryFile(dir=get_downloads_path()) as named_temp_file:
+                download_url = self.tarball
+
+                download_untar_fn = functools.partial(
+                    self.download_tar
+                    , download_url
+                    , named_temp_file.name
+                )
+                connection_exception_retry(download_untar_fn, 5)
+
+                validate_checksum(self, named_temp_file.name)
+                validate_tarfile(self, named_temp_file.name)
+                tar = tarfile.open(named_temp_file.name, "r:gz")
+                self.tarfile_bin = tar
+
+        if type(self.tarball) is str:
+            if os.path.isfile(self.tarball):
+                print('handle_local_tar')
+                handle_local_tar(self)
+            else:
+                # assume download if str and not local file
+                print('handle_remote_tar')
+                handle_remote_tar(self)
+        elif type(self.tarball) is tarfile.TarFile:
+            # pass through TarFile object primarily for testing
+            # no explicit hash test here
+            self.tarfile_bin = self.tarball
+        else:
+            raise ValueError('unknown tarball type defined')
+
+        if not self.get_tar_size() <= TARFILE_MAX_SIZE:
+            msg = ("Tarfile size is larger than limit "
+                   f"of {TARFILE_MAX_SIZE}.")
+            raise DependencyException(msg)
+
+            fire_event(Sha1ChecksumPasses())
```

Comment:
This (and the sha1 checks) were removed as of [25ee2c6](https://github.com/dbt-labs/dbt-core/pull/4689/commits/25ee2c69712186dc3184fa966b66a318bd82e37b). There's no matches for `fire_event(Sha1ChecksumPasses())` in the current revision. Looks like this may just be based on an outdated snippet, but let me know if I've missed something!